src/Chapter2.hs

```diff
@@ -605,8 +629,8 @@ Implement a function that duplicates each element of the list
 
 -}
 duplicate :: [a] -> [a]
-duplicate = error "duplicate: Not implemented!"
-
+duplicate [] = []
+duplicate (x:xs) = [x,x] ++ duplicate xs
```

Comment:
That's right! 👍🏼 
In general, it's always a good idea to try to replace `++` with `x : l` wherever possible. Remember the analogy with trains :bullettrain_side: :train: :train: :train:
In here, you can just `x` into the beginning of the list with the `:` operator:
```suggestion
duplicate (x:xs) = x : x : duplicate xs
```